### PR TITLE
Add AES-SIV (RFC 5297).

### PR DESCRIPTION
**Pull request by haydenroche5**
This commit adds functions to encrypt and decrypt data using AES in SIV mode, as
described in RFC 5297. This was added in the process of porting chrony to
wolfSSL. chrony is an NTP implementation that can use NTS (network time
security), which requires AES-SIV.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -1590,14 +1590,20 @@ then
     AM_CFLAGS="$AM_CFLAGS -DHAVE_AESCCM"
 fi
 
+# AES-SIV (RFC 5297)
```
Comment (dgarske) [inline configure.ac:1593]: Please add `enable_aessiv" to the `[all-crypto]` section. Thanks

**Review by dgarske — CHANGES_REQUESTED** [review]
Please add AES SIV to the benchmarks. Thanks

**Review by dgarske — APPROVED** [review]

File: wolfcrypt/src/misc.c [hunk 2]
```diff
@@ -546,6 +546,30 @@ WC_STATIC WC_INLINE byte ctSetLTE(int a, int b)
 }
 #endif
 
+#ifndef HAVE_FIPS
+/* This function used to be local to cmac.c. It was moved here because the
+ * AES-SIV code also needs it. cmac.c is part of the FIPS module, though, so we
+ * need to not compile this definition if using FIPS. */
+WC_STATIC WC_INLINE void ShiftAndXorRb(byte* out, byte* in, word32 sz)
```
Comment (SparkiDev) [inline wolfcrypt/src/misc.c:553]: misc.c is for general functions like XOR buffers.
This seems algorithm specific.
I would like it in cmac.c as a local API or as a #define in a header file.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: wolfcrypt/src/misc.c (hunk 2 above)
Comment (haydenroche5) [reply wolfcrypt/src/misc.c:553]: That makes sense. I can make it `WOLFSSL_LOCAL` and part of cmac.c/h, and then it can be used in aes.c.

**Review by SparkiDev — APPROVED** [review]

**Review by dgarske — APPROVED** [review]

**Comment by dgarske** [comment]
Please resolve merge conflict and happy to merge. Thanks @haydenroche5

**Pull request merged**